packages/@aws-cdk/aws-kinesis/lib/stream.ts

```diff
@@ -752,9 +755,15 @@ export class Stream extends StreamBase {
       physicalName: props.streamName,
     });
 
-    const shardCount = props.shardCount || 1;
+    let shardCount = props.shardCount;
+    let streamMode = props.streamMode ?? StreamMode.PROVISIONED;
 
-    const streamMode = props.streamMode;
+    if (streamMode === StreamMode.ON_DEMAND && shardCount !== undefined) {
+      throw new Error('streamMode must be set to PROVISIONED (default) when specifying shardCount');
+    }
```

Comment:
Could be an else if

Reply:
This is a matter of personal preference. Adding `else` here doesn't make this code any clearer to me so I prefer not to.